packages/tools/devtools/devtools-view/src/components/data-visualization/EditableValueView.tsx

```diff
@@ -0,0 +1,134 @@
+import { TreeHeader } from "./TreeHeader";
+
+/**
+ * Input to {@link EditableValueView}
+ */
+export interface EditableValueViewProps {
+	node: FluidObjectValueNode;
+	containerKey: string;
+	label: string;
+}
+
+/**
+ * Constructs a editable text field to allow editing of a DDS' content.
+ */
+export function EditableValueView(props: EditableValueViewProps): React.ReactElement {
+	const { node, containerKey, label } = props;
+	const messageRelay = useMessageRelay();
+
+	/**
+	 * value is the current value of the text field.
+	 */
+	const [value, setValue] = React.useState<string>(String(node.value));
+
+	/**
+	 * isEditing is whether or not the user is currently attempting to edit a value. If they are it will not update so their changes are not overwritten
+	 */
+	const [isEditing, setIsEditing] = React.useState<boolean>(false);
+
+	/**
+	 * isType declares what type the edit will be
+	 */
+	const [isType, setIsType] = React.useState<string>(typeof node.value);
+
+	const textAreaRef = React.useRef<HTMLInputElement>(null);
+
+	const backgroundUpdate = (): void => {
+		if (isEditing === false) {
+			setValue(String(node.value));
+			setIsType(typeof node.value);
+		}
+	};
+	React.useEffect(backgroundUpdate, [node.value, isEditing]);
+
+	const onFocus = React.useCallback(() => {
+		if (textAreaRef.current !== null) {
+			textAreaRef.current?.select();
+		}
+
+		setIsEditing(true);
+	}, []);
+
+	const onChange = React.useCallback(
+		(_ev: React.ChangeEvent<HTMLInputElement>, data: InputOnChangeData) => {
+			setValue(data.value);
+		},
+		[],
+	);
+
+	const onBlur = React.useCallback(() => {
+		setValue(String(node.value));
+		setIsEditing(false);
+	}, [node.value]);
+
+	const commitChanges = React.useCallback(() => {
+		let data: Serializable<unknown> = value;
+		switch (isType) {
+			case "number":
+				data = Number(value);
+				break;
+			case "boolean":
+				data = value === "true" ? true : false;
+				break;
+			default:
+				data = String(value);
+				break;
+		}
+		const edit = {
+			fluidObjectId: node.fluidObjectId,
+			data,
+		};
+		messageRelay.postMessage(
+			SendEditData.createMessage({
+				containerKey,
+				edit,
+			}),
+		);
+	}, [containerKey, isType, messageRelay, node.fluidObjectId, value]);
+
+	const onKeyDown = React.useCallback(
+		(event: React.KeyboardEvent<HTMLInputElement>) => {
+			const key = event.key;
+
+			if (key === "Enter") {
+				commitChanges();
+				event.currentTarget.blur();
+			}
+
+			if (key === "Escape") {
+				event.currentTarget.blur();
+			}
+		},
+		[commitChanges],
+	);
+
+	return (
+		<>
+			<TreeHeader label={label} nodeTypeMetadata={node.typeMetadata}></TreeHeader>
+			<Input
+				size="small"
+				appearance="underline"
+				contentEditable
+				ref={textAreaRef}
+				onClick={(event): void => event.preventDefault()}
+				value={String(value)}
+				onChange={onChange}
+				onFocus={onFocus}
+				onBlur={onBlur}
+				onKeyDown={onKeyDown}
+				type={isType === "string" ? "text" : "number"}
```

Comment:
Nit: Currently, if we see anything other than "number" or "string", this will set the control to "number". Might be good to extract this into a function and explicitly handle (or error) if we encounter something other than "number" or "string".

Reply:
I exatracted to a function. I will add some default case to it along with some chagnes that need to be implemented with sharedCell dropdown UI. Could I push off the error handling to the next PR?

Reply:
Sorry, I only meant extracting the mapping from `editType` to string ("text", "number", etc.) into a function. Mapping to different controls / components can come later IMO. Let me know if that doesn't make sense.

Reply:
I was getting some strange problems where if simply giving the type prop a "string" would give errors, even if that string would only ever be valid for it. This was the way I had thought of getting around that and potentially saving some work going into forward. If this doesn't match with what you were thinking/suggesting I need more details.

Reply:
I can look at this with you, if you'd like.